src/Pages/homePage/homePage.js

```diff
@@ -1,8 +1,91 @@
-import React from "react";
+import React, {useEffect, useState} from "react";
+import {useDebounce} from "use-debounce";
+import {apiUrl} from "../../config";
+import orderBy from "lodash/orderBy";
+import { Input } from "../../components/Input";
+
+
+
 
 export const HomePage = () => {
+    const [products, setProducts] = useState([]);
+    const [selectedSearch] = useState("");
+    const [selectedOrderBy, setSelectedOrderBy] = useState("");
+    //TODO: get possible orders from database, MP 08/05
+
+    const orderByOptions = [
+        {label: "Name", value: "name"},
+        {label: "Note", value: "note"},
+        {label: "Expiry", value: "expiryDate"},
+    ];
+
+    const [debouncedSearch] = useDebounce(selectedSearch, 300);
+
+    console.log(products);
+
+
+
+    useEffect(() => {
+        // Fetching all products
+        fetch(apiUrl + "/products")
+            .then((res) => res.json())
+            .then((res) => {
+                if (res.success) {
+                    // Set new categories
+                    setProducts(res.data);
+                }
+            });
+    }, []);
+
+    useEffect(() => {
+        const newOrder = selectedOrderBy || "name";
+        setProducts(orderBy(products, [newOrder]));
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selectedOrderBy]);
+
+    useEffect(() => {
+        // fail check
+        const parametersArray = [];
+
+        if (debouncedSearch) {
+            parametersArray.push("search=" + encodeURIComponent(debouncedSearch));
+        }
+        const endpoint = "/products?" + parametersArray.join("&");
+
+        fetch(apiUrl + endpoint)
+            .then((res) => res.json())
+            .then((res) => {
+                if (res.success) {
+                    setProducts(res.data);
+                    setSelectedOrderBy("");
+                }
+            });
+    }, [debouncedSearch]);
+
+    Date.prototype.addDays = function (days) {
```

Comment:
This is a way to add custom things to classes. However, React is trying to move from OOP to Functional Programming. I would have probably used a simple library which can format days

Reply:
I will look into that